### PR TITLE
Gulp pipeable - fixes #41

### LINKED ISSUE #41: Should be pipe-able in a gulp chain, and the task helper should allow optional dependencies
The library should be able to sit in between `gulp.src` and `gulp.dest` streams - just like other gulp plugins:

```javascript
var compiler = require('can-compile/gulp');

gulp.task('cancompile', function () {
   return gulp.src('path/to/templates')
       .pipe( compiler('views.production.js') )
       .pipe( gulp.dest('path/to/dist') );
});
```

Also, the `.task` helper should allow for optional dependencies (notice `[clean]`):

```javascript
var compiler = require('can-compile/gulp');
compiler.task('cancompile', {
    version: '2.2.7',
    src: 'glob/to/source/*/**'
    out: 'path/to/dest/views.templates.js'
}, gulp, ['clean']);
```

### PR DESCRIPTION
**Pull request by DesignByOnyx**
This should contain all the latest and greatest now!

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -9,27 +9,6 @@ With NodeJS installed, just run NPM:
 
 > npm install can-compile -g
 
-## Command line
```
Comment (daffl) [inline readme.md]: Why is this moved down? I would like people to read about the command line usage before talking about build tools.

Comment (DesignByOnyx) [inline readme.md]: I guess I would like people to read about the gulp/grunt tasks before the command line. Just a difference of preference. I can move it back up if you like.

Comment (daffl) [inline readme.md]: Yes please. I think it is important that it works standalone (I personally barely use any build tools anymore because most tools now have a great CLI).
